WeatherApp/Model/WeatherManager.swift

```diff
@@ -10,95 +10,55 @@ import Foundation
         let urlString = "\(rawURL)&lat=\(latitude)&lon=\(longitude)"
         performWeatherRequest(with: urlString, fromLocation: true)
     }
-    
-    fileprivate func prepareNameToRequest(for cityName: String) -> String {
-        
+
+    private func prepareNameToRequest(for cityName: String) -> String {
+
         let removedSpaces = cityName.trimmingCharacters(in: .whitespacesAndNewlines).replacingOccurrences(of: " ", with: "%20")
         let removedDiacritics = removedSpaces.lowercased().folding(options: .diacriticInsensitive, locale: .current).replacingOccurrences(of: "ł", with: "l")
-        
+
         return removedDiacritics
     }
-    
-    func performWeatherRequest(with url: String, fromLocation: Bool) {
-        Alamofire.request(url).response { (responseData) in
-            if let weatherData = responseData.data {
-                guard var weatherModel = self.parseJSON(with: weatherData) else {
-                    return
-                }
-                
-                weatherModel.fromLocation = fromLocation
-                self.delegate?.weatherDataDidUpdate(self, weather: weatherModel)
+
+    private func performWeatherRequest(with url: String, fromLocation: Bool) {
+        Alamofire.request(url).response { (response) in
+            guard let data = response.data else {
+                return
             }
-        }
-    }
-    
-    func parseJSON(with weatherData: Data) -> WeatherModel? {
-        let data = JSON(weatherData)
-        
-        if data["cod"].stringValue != "200" {
-            return nil
-        }
-        
-        let cityNameValue = data["city"]["name"].stringValue
-        
-        var dayForecast: [WeatherModel.DayForecast] = []
-        
-        data["list"].arrayValue.forEach { (day) in
-            let temp = day["main"]["temp"].intValue
-            let condId = day["weather"][0]["id"].intValue
-            let description = day["weather"][0]["description"].stringValue
-            let dateTime = day["dt_txt"].stringValue.split(separator: " ")
-            let date = reformatDate(in: String(dateTime[0]))
-            let time = reformatTime(in: String(dateTime[1]))
-            
-            dayForecast.append(WeatherModel.DayForecast(conditionID: condId, temp: temp, description: description, date: date, time: time))
-        }
-        
-        return WeatherModel(cityName: cityNameValue, dayForecast: dayForecast, fromLocation: false)
-    }
-    
-    func reformatDate(in value: String) -> String {
-        let dateFormatterGet = DateFormatter()
-        dateFormatterGet.dateFormat = "yyyy-MM-dd"
 
-        let dateFormatter = DateFormatter()
-        dateFormatter.dateFormat = "E, d MMM"
+            guard let model = self.mapDataToModel(data, fromLocation: fromLocation) else {
+                return
+            }
 
-        let date: Date? = dateFormatterGet.date(from: value)
-        
-        return dateFormatter.string(from: date!)
```

Comment:
Wykrzykników (force unwrap) unikamy jak ognia, bo apka crashuje nawet nie wiesz kiedy :D To samo się tyczy wykrzykników przy castowaniu (force cast) np. `let value = someProperty as! SomeType`.